system/HandlingBlockchainDowntime.md

```diff
@@ -0,0 +1,42 @@
+# Handling blockchain downtime
+
+2021-10-06 was a hard day for Mysterium while running in testnet3. The polygon mumbai blockchain maintainers decided to do a release to validators, which in turn broke the whole mumbai testnet. Since we were running on mumbai at that time, none of our services worked. Nodes failed to start, the running nodes could not establish/maintain sessions as Hermes would encounter issues when checking the channel status for both providers and consumers on chain.
+
+## Learning from mistakes of the past
+
+To prevent such issues, a few fixes need to be made:
+
+## The proposed fixes
+
+### Node
+
+Node failed to start due to the fact that they could not lookup hermes address(URL) via blockchain. This could be solved by adding a fallback to the hermes url for known hermeses directly in node.
+
+However, this is unlikely to be a sufficient step to solve the startup issues as I believe there are many more further checks that happen on startup that rely on blockchain.
+
+From memory, some further lookups that might be optional are:
+
+1) On session start, provider checks the hermes fee via blockchain. This can easily be cached for reasonably long times if it already is not.
+1) On session start, provider checks the hermes status via blockchain. This call should probably be remade in a way where a failure to call blockchain is not treated as critical.
+
+All of them can likely be done via Hermes + Transactor instead of blockchain as long as those two services are alive. 
+
+Ideally, one would add full node URLs to ones' hosts file and point them to localhost, try to start node and not rest until all the issues are solved.
```

Comment:
I was more referring to the way of debugging what is critical for node startup. 

Having a light node is possible, even on mobile, but the issue is that it won't help in case of total failure like the one we had.

Reply:
You mean you propose calling Hermes and transactor instead of blockchain? I would disagree, node should not fully rely on hermes as hermes should be trustless service.

However in emergency situation, where there is no connection to blockchain, we may use it as fallback solution.